File: binary_search.java
import java.util.Scanner;

public class binary_search{

  public static void main(String args[]){
    Scanner sc = new Scanner(System.in);
    System.out.println("Enter the number of elements you want to enter:");
    int n = sc.nextInt();
    int list[] = new int[n];
    for(int i = 0; i < n; i++){
      System.out.println("enter number:");
      list[i] = sc.nextInt();
    }
    System.out.println("enter number to search for:");
    int search = sc.nextInt();
    int min = 0;
    int max = n-1;
    boolean found = false;
    while(min < max && !found){
      int mid = (min + max)/2;
      if(list[mid] > search){
        max = mid - 1;
        System.out.println(list[mid]);
      }else if(list[mid] < search){
        min = mid + 1;
        System.out.println(list[mid]);
      }else{
        found = true;
        System.out.println("found at:" + (mid+1));
      }
    }
    if(!found){
      System.out.println("number not found");
    }
  }
}

File: binary_search.py
list = [1,3,5,7,9,12,34,56,67,71,72,73,86,87,89,92,94,97,100]

def binary_search(list, search):
    min = 0;
    max = len(list) - 1
    found = False
    while min < max and (not found):
        mid = int( (min + max)/2)
        if list[mid] > search:
            max = mid - 1
        elif list[mid] < search:
            min = mid + 1
        else :
            return mid + 1
    return -1

x = binary_search(list,34)
print( "found at position:"+ str(x))
